### PR TITLE
cpp-httplib: add version 0.18.2

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **cpp-httplib/0.18.2**

#### Motivation
There are several bugfixes in 0.18.2.

#### Details
https://github.com/yhirose/cpp-httplib/releases/tag/v0.18.2
https://github.com/yhirose/cpp-httplib/compare/v0.18.1...v0.18.2

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/cpp-httplib/all/conanfile.py [hunk 1]
```diff
@@ -20,17 +21,27 @@ class CpphttplibConan(ConanFile):
...
+            del self.options.use_macos_keychain_certs
+
     def requirements(self):
         if self.options.with_openssl:
-            self.requires("openssl/[>=1.1 <4]")
+            if Version(self.version) < "0.15":
+                self.requires("openssl/[>=1.1 <4]")
+            else:
+                # New version of httplib.h requires OpenSSL 3
+                self.requires("openssl/[>=3 <4]")
```
Comment (AbrilRBS) [inline recipes/cpp-httplib/all/conanfile.py:44]: Activating openssl for newer versions triggered 
`httplib.h:273:2: error: Sorry, OpenSSL versions prior to 3.0.0 are not supported` for all versions after 0.15

File: recipes/cpp-httplib/all/conanfile.py [hunk 2]
```diff
@@ -20,17 +21,27 @@ class CpphttplibConan(ConanFile):
         "with_openssl": [True, False],
         "with_zlib": [True, False],
         "with_brotli": [True, False],
+        "use_macos_keychain_certs": [True, False],
     }
     default_options = {
         "with_openssl": False,
         "with_zlib": False,
         "with_brotli": False,
+        "use_macos_keychain_certs": True,
```
Comment (AbrilRBS) [inline recipes/cpp-httplib/all/conanfile.py:30]: Won't change package id as it's a header-only library

**Review by AbrilRBS — APPROVED** [review]
Thanks! I've done some fixes and cleanups

**Review by uilianries — APPROVED** [review]
LGTM. Good catch with exposed flags when using this new option.

The option is available before 0.14.1: https://github.com/yhirose/cpp-httplib/blob/v0.14.1/CMakeLists.txt#L9

**Pull request merged**